include/highfive/h5easy_bits/H5Easy_scalar.hpp

```diff
@@ -16,27 +16,34 @@ namespace H5Easy {
 
 namespace detail {
 
-namespace scalar {
+/*
+This base class presents fallback methods which will be automatically
+called from specializations, if not defined in the specialization.
+Every specialization of io_impl<T> should inherit from io_impl_base<T>.
```

Comment:
I don't think this is entirely true, at least not for the moment. If you call `load_part` with e.g. an Eigen object you will get a compiler error because `load_part` is not in the `struct` enabled for `Eigen` objects. I don't see this as a problem a-priori: the methods could be added if so desired. For the moment defining the methods and putting a `static_assert` might improve readability.

Reply:
As far as I understood, `std::vector` uses some methods for "scalars" directly, this makes it possible; otherwise they would not be accessible, would have to be duplicated.

Reply:
I think is a reply to the next comment. (The one here is I think still important to discuss). My point was that the `io_impl` of `std::vector<T>` will be enabled when called with a `std::vector<T>` so I guess any function in `H5Easy_scalar` can not be reached with any `std::vector<T>`